Fix for new subscription dialog

ashwini-seshadri opened the pull request:
#### What? Why?

Closes #9648

When setting up a new subscription the list of shop open and shows a search input but was not able to type anything in there. This PR fixes that and there is no need for the user to scroll through the list of shops to find the one they need.




#### What should we test?
Other parts of the app will not be affected from this change.

#### Release notes

<!-- Please select one for your PR and delete the other. -->

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.


#### Dependencies
<!-- Does this PR depend on another one?
     Add the link or remove this section. -->
No


#### Documentation updates
<!-- Are there any wiki pages that need updating after merging this PR?
     List them here or remove this section. -->
No

ashwini-seshadri commented:
https://user-images.githubusercontent.com/88281470/192154016-029939f8-93b5-47c0-adad-44da5e3656ec.mov

ashwini-seshadri commented:
![Screenshot 2022-09-25 at 6 19 52 PM](https://user-images.githubusercontent.com/88281470/192154072-372f37ee-9c14-4175-b3fe-d9a662a1741e.png)

I see a "triangle" in green color from the nav bar showing up on the modal. Shall I increase the z-index of the modal?

ashwini-seshadri commented on spec/support/request/web_helper.rb:
```diff
@@ -128,9 +128,11 @@ def close_select2
-    page.find(:css, '.ts-dropdown input.dropdown-input').set(value)
-    page.find(:css, '.ts-dropdown .ts-dropdown-content .option', text: value).click
+    tomselect_wrapper = page.find("[name='#{options[:from]}']").sibling(".ts-wrapper")
+    tomselect_wrapper.find(".ts-control").click
```
tom select along with capybara was behaving weird, and I was seeing this failure. So I did multiple clicks. I hope it is ok.:
```
Capybara::ElementNotFound:
       Unable to find visible css ".ts-dropdown input.dropdown-input" within #<Capybara::Node::Element tag="div" path="//HTML[1]/BODY[1]/DIV[1]/DIV[1]/DIV[4]/DIV[1]/DIV[1]/DIV[1]/DIV[2]/DIV[2]/DIV[1]/FORM[1]/DIV[1]/DIV[1]">
```

ashwini-seshadri commented on spec/support/request/web_helper.rb:
```diff
@@ -128,9 +128,11 @@ def close_select2
-    page.find("[name='#{options[:from]}']").sibling(".ts-wrapper").click
-    page.find(:css, '.ts-dropdown input.dropdown-input').set(value)
-    page.find(:css, '.ts-dropdown .ts-dropdown-content .option', text: value).click
+    tomselect_wrapper = page.find("[name='#{options[:from]}']").sibling(".ts-wrapper")
```
I think we have to do this instead of full page, since a page could contain multiple tom select elements and we might end up targeting wrong element.

jibees replied:
Yep. I think you're right.

mkllnk reviewed (approved):
Thank you! The code looks good to me. Let's see what @jibees says.

jibees commented on app/views/admin/subscriptions/index.html.haml:
```diff
@@ -6,7 +6,7 @@
 - content_for :page_actions do
   %li{ "data-controller": "help-modal-link", "data-action": "click->help-modal-link#open", "data-help-modal-link-target-value": "new_subscription_help_modal" }
-    %a.button.icon-plus#new-subscription
+    %a.button.icon-plus#new-subscription{ href: "javascript:void(0)" }
```
Not sure to understand this change?

ashwini-seshadri replied:
The capybara spec was unable to target the link.

mkllnk replied:
We shouldn't change the code for no other benefit than making a spec work. I could imagine that Capybara expects an `href` to use `click_link`. But could you use `find(...).click`?

jibees reviewed (changes requested):
Quick comments below on commits, and general comments:

 - I think you should at least delete the second commit which is a bad merge commit, and probably squash the two commits into one because the last is only "fix specs".
 - Outside purely code review, I see : 
   - you mention a green arrow. Where does it comes? Sounds like they should not appear here right?
   - I think that text is center aligned in the dropdown. I don't think it's the way to go, as in other place, the text is left aligned.

ashwini-seshadri commented:
> Quick comments below on commits, and general comments:
> 
> * I think you should at least delete the second commit which is a bad merge commit, and probably squash the two commits into one because the last is only "fix specs".
> * Outside purely code review, I see :
>   
>   * you mention a green arrow. Where does it comes? Sounds like they should not appear here right?
>   * I think that text is center aligned in the dropdown. I don't think it's the way to go, as in other place, the text is left aligned.

Hi, @jibees I have squashed all the commits into a single commit.
I have changed the text alignment to left in the dropdown.
I have also fixed the green arrow by changing it's z-index to 0 which was coming from [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/9702/files#diff-80b53678d55024c2f3c7aeb1b19f22f04b281182b214545137be31f6a152ee8cR161)

jibees reviewed (approved):
Nice 💪 
Thanks a lot @ashwini-seshadri !
Moving to `test ready`

filipefurtad0 commented:
Hey @ashwini-seshadri ,
Welcome :wave: 

Thanks for picking this up, I've verified the issue prior to staging and checked that your PR fixes it after staging it:

[Screencast from 30-09-2022 13:29:30.webm](https://user-images.githubusercontent.com/49817236/193270291-803d3943-3fbc-4f5b-aa33-5e803b05b934.webm)

Many thanks! Merging.

drummer83 commented:
@ashwini-seshadri I have just used this new functionality for the first time! It's amazing!
Thank you again! :tada:
